apps/minifront/src/state/swap/price-history.ts

```diff
@@ -1,47 +1,73 @@
 import { Metadata } from '@buf/penumbra-zone_penumbra.bufbuild_es/penumbra/core/asset/v1/asset_pb.js';
-import { CandlestickData } from '@buf/penumbra-zone_penumbra.bufbuild_es/penumbra/core/component/dex/v1/dex_pb.js';
-import { getMetadataFromBalancesResponseOptional } from '@penumbra-zone/getters/balances-response';
-import { AllSlices, SliceCreator } from '..';
-import { sendCandlestickDataRequests } from './helpers';
+import { CandlestickDataResponse } from '@buf/penumbra-zone_penumbra.bufbuild_es/penumbra/core/component/dex/v1/dex_pb.js';
 import { PRICE_RELEVANCE_THRESHOLDS } from '@penumbra-zone/types/assets';
+import { createZQuery, ZQueryState } from '@penumbra-zone/zquery';
+import { AllSlices, SliceCreator, useStore } from '..';
+import { sendCandlestickDataRequest } from './helpers';
 
 interface Actions {
-  load: (ac?: AbortController) => AbortController['abort'];
+  /**
+   * History limit becomes the maximum width of the chart domain (block height).
+   */
+  setHistoryLimit: (limit: bigint) => void;
+  /**
+   * Setting history start will cause the chart domain to begin at the specified
+   * block height and extend towards the present. Setting history start to
+   * `undefined` or `0n` will cause the chart domain to end at the present block
+   * height and extend towards the past.
+   */
+  setHistoryStart: (blockHeight?: bigint) => void;
 }
 
+export const { candles, useCandles, useRevalidateCandles } = createZQuery({
+  name: 'candles',
+  fetch: (
+    startMetadata: Metadata | undefined,
+    endMetadata: Metadata | undefined,
+    historyLimit: bigint | undefined,
+    historyStart: bigint | undefined,
+  ) =>
+    Promise.all([
+      sendCandlestickDataRequest(startMetadata, endMetadata, historyLimit, historyStart),
+      sendCandlestickDataRequest(endMetadata, startMetadata, historyLimit, historyStart),
+    ]).then(([direct, inverse]) => ({ direct, inverse })),
```

Comment:
suggestion: make this a named function import

Not a requirement, but it's a little cleaner to have something like:
```suggestion
  fetch: getDirectAndInverseCandlestickData,
```

...and thus keep logic out of the ZQuery object definition.